contracts/strategies/looksrare/LooksRareStakingStrategy.sol

```diff
@@ -0,0 +1,245 @@
+  */
+  function continueInvesting() public onlyGovernance {
+    _setPausedInvesting(false);
+  }
+
+  function _liquidateReward() internal {
+    uint256 rewardBalance = IERC20(rewardToken()).balanceOf(address(this));
+    if (!sell() || rewardBalance < sellFloor()) {
+      // Profits can be disabled for possible simplified and rapid exit
+      emit ProfitsNotCollected(sell(), rewardBalance < sellFloor());
+      return;
+    }
+
+    notifyProfitInRewardToken(rewardBalance);
+    uint256 remainingRewardBalance = IERC20(rewardToken()).balanceOf(address(this));
+
+    if (remainingRewardBalance <= 0) {
+      return;
+    }
+
+    IERC20(rewardToken()).safeApprove(universalLiquidator(), 0);
+    IERC20(rewardToken()).safeApprove(universalLiquidator(), remainingRewardBalance);
+
+    // we can accept 1 as minimum because this is called only by a trusted role
+    ILiquidator(universalLiquidator()).swapTokenOnMultipleDEXes(
+      remainingRewardBalance,
+      1,
+      address(this), // target
+      storedLiquidationDexes[rewardToken()][underlying()],
+      storedLiquidationPaths[rewardToken()][underlying()]
+    );
+  }
+
+  /*
+  *   Stakes everything the strategy holds into the reward pool
+  */
+  function investAllUnderlying() internal onlyNotPausedInvesting {
+    if(IERC20(underlying()).balanceOf(address(this)) > 0) {
+      enterRewardPool();
+    }
+  }
+
+  /*
+  *   Withdraws all the asset to the vault
+  */
+  function withdrawAllToVault() public restricted {
+    if (address(rewardPool()) != address(0)) {
+      exitRewardPool();
+    }
+    _liquidateReward();
+    IERC20(underlying()).safeTransfer(vault(), IERC20(underlying()).balanceOf(address(this)));
+  }
+
+  /*
+  *   Withdraws the amount for the asset to the vault
+  */
+  function withdrawToVault(uint256 amount) public restricted {
+    // Typically there wouldn't be any amount here
+    // however, it is possible because of the emergencyExit
+    uint256 entireBalance = IERC20(underlying()).balanceOf(address(this));
+
+    if(amount > entireBalance){
+      // While we have the check above, we still using SafeMath below
+      // for the peace of mind (in case something gets changed in between)
+      uint256 needToWithdrawLOOKS = amount.sub(entireBalance);
+
+      uint256 canWithdrawLOOKS = Math.min(rewardPoolBalance(), needToWithdrawLOOKS);
+
+      // convert amount of underlying to withdraw to shares price
+      uint256 oneShareInLOOKS = IFeeSharingSystem(rewardPool()).calculateSharePriceInLOOKS();
+      uint256 sharesToWithdraw = canWithdrawLOOKS.div(oneShareInLOOKS);
+
+      // bool flag param for claiming reward tokens set to false
+      IFeeSharingSystem(rewardPool()).withdraw(sharesToWithdraw, false);
+    }
+
+    uint256 underlyingBalance = IERC20(underlying()).balanceOf(address(this));
+
+    // recalculate to improve accuracy
+    uint256 underlyingAmountToWithdraw = Math.min(amount, underlyingBalance);
```

Comment:
I'm not sure we would want this to be here. If for whatever reason we were only to withdraw less underlying from the strategy this would make it go through with the lesser amount. We might just want it to fail if the available amount is less than the requested amount. What do you think?

Reply:
true, removed it.

Reply:
The thing is that the shares will not equal looks 100% exactly due to precision issues. 

In the tests it runs fine because the underlyingBalance ends up being more, but I am not sure if it might be possible that the underlying balance is slightly below the amount to withdraw.

![image](https://user-images.githubusercontent.com/6112929/150517102-b233217d-0b4b-47da-88f6-2e16a24baa27.png)

What if I'd add a require that checks if the underlying balance is within a certain tolerance of the amount to withdraw, e.g.
last 6 to 7 digits of the 1e18 precision amount? 
If the require passes then the withdrawal should pass with the new underlyingBalance as transferAmount.

That would ensure we don't run into weird issues potentially?

Reply:
Would need to check if this is good enough for the further processing in the vault. If an exact amount (or a strict minimum) is expected to come into the vault in handling a withdrawal in the vault contract we still have a problem I think.

Reply:
The vault actually recalculates too, that's where I had the idea from :smile: 

https://github.com/harvest-finance/harvest-strategy/blob/master/contracts/base/VaultV3.sol#L294

https://github.com/harvest-finance/harvest-strategy/blob/master/contracts/base/VaultPausable.sol#L288

Reply:
The alternative of course would be to just ensure enough LOOKS are withdrawn by adding a small tolerance amount. That would probably be better. 
And having a tiny bit of dust around in the strategy until the next action shouldn't be a problem

Reply:
Sounds good!

Reply:
Great! Please resolve this issue if you reviewed the logic that I added, thank you!

Reply:
I'm not actually sure how to do that lol. But the changes look good!